File: server/BeerWorkshop.Application/Dto/Responses/Statistic/StatisticProductRowResponseDto.cs
using BeerWorkshop.Database.Enums;

namespace BeerWorkshop.Application.Dto.Responses.Statistic;

public record StatisticProductRowResponseDto(
    string Name,
    ProductType ProductType,
    TransactionType TransactionType,
    decimal Quantity,
    decimal Price,
    string? Discount,
    decimal TotalAmount,
    DateTime TransactionDate
);

File: client/src/redux/dtos/responses/statistic/statisticProductRowResponseDto.ts
import { ProductType, TransactionType } from "../../../enums";

export interface StatisticProductRowResponseDto {
  name: string;
  productType: ProductType;
  transactionType: TransactionType;
  quantity: number;
  price: number;
  discount: string;
  totalAmount: number;
  transactionDate: Date;
}
